lib/bokken/accounts.ex

```diff
@@ -688,4 +688,253 @@ defmodule Bokken.Accounts do
   def change_user(%User{} = user, attrs \\ %{}) do
     User.changeset(user, attrs)
   end
+
+  @doc """
+  Gets a single skill.
+
+  Raises `Ecto.NoResultsError` if the Skill does not exist.
+
+  ## Examples
+
+      iex> get_skill!(123)
+      %Skill{}
+
+      iex> get_skill!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_skill!(id) do
+    Repo.get!(Skill, id)
+  end
+
+  @doc """
+  Returns the list of skills.
+  ## Examples
+      iex> list_skills()
+      [%Skill{}, ...]
+  """
+  def list_skills do
+    Skill
+    |> Repo.all()
+  end
+
+  @doc """
+  Creates a skill.
+
+  ## Examples
+
+      iex> create_skill(%{field: value})
+      {:ok, %Skill{}}
+
+      iex> create_skill(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def create_skill(attrs \\ %{}) do
+    %Skill{}
+    |> Skill.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  Updates a skill.
+
+  ## Examples
+
+      iex> update_skill(%Skill{}, %{field: value})
+      {:ok, %Skill{}}
+
+      iex> update_skill(%Skill{}, %{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def update_skill(%Skill{} = skill, attrs \\ %{}) do
+    skill
+    |> Skill.changeset(attrs)
+    |> Repo.update()
+  end
+
+  @doc """
+  Deletes a  skill.
+
+  ## Examples
+
+      iex> delete_skill(skill)
+      {:ok, %Skill{}}
+
+      iex> delete_skill(skill)
+      {:error, %Ecto.Changeset{}}
+  """
+  def delete_skill(%Skill{} = skill) do
+    Repo.delete(skill)
+  end
+
+  @doc """
+  Returns an `%Ecto.Changeset{}` for tracking skill changes.
+
+  ## Examples
+
+      iex> change_skill(skill)
+      %Ecto.Changeset{data: %Skill{}}
+
+  """
+  def change_skill(%Skill{} = skill, attrs \\ %{}) do
+    Skill.changeset(skill, attrs)
+  end
+
+  @doc """
+  Returns whether or not the ninja has a Skill
+
+  ## Examples
+
+      iex> ninja_has_skill?(123, 123)
+      true
+
+      iex> ninja_has_skill?(123, 456)
+      false
+
+  """
+  def ninja_has_skill?(ninja_id, skill_id) do
+    NinjaSkill
+    |> where([ns], ns.ninja_id == ^ninja_id and ns.skill_id == ^skill_id)
+    |> Repo.exists?()
+  end
+
+  @doc """
+  Returns the list of skills of a ninja.
+  ## Examples
+      iex> list_ninja_skills(123)
+      [%Skill{}, ...]
+  """
+  def list_ninja_skills(ninja_id) do
```

Comment:
I think these functions would have a better api if they worked like:


```elixir
def list_skills(%{"ninja_id" => ninja_id}) do
   NinjaSkill
    |> where([ns], ns.ninja_id == ^ninja_id)
    |> join(:inner, [ns], s in Skill, on: s.id == ns.skill_id)
    |> select([ns, s], s)
    |> Repo.all()
 end 
 ```

Reply:
Why?

Reply:
That makes it more consistent with other listing functions we use and in the controller you could just pass the parameters directly, making it one less place to change things if needed.